MetaMask/metamask-mobile
# [FEATURE] SDK Session Persistence

**Description**

Refactor of the SDKConnect class with added MM SDK "session persistence" feature.

SDK Session persistence allows a dApp to automatically reconnect a user to his MetaMask wallet without having to re-approve the connection. This is especially useful to improve the user experience when a user navigate.

New UI to manage SDK sessions moved from "Experimental" to "Advanced" settings of the app.

Added OTP management to secure dApp connection from QR-Code origin.

Current PR includes a copy of the MM SDK `communication-layer` protocol as the npm package hasn't been published yet.

**Screenshots/Recordings**
![image](https://user-images.githubusercontent.com/107169956/224690320-5acf7163-cb13-4daf-8371-b5cbaa628a7e.png)
![image](https://user-images.githubusercontent.com/107169956/224690438-1404e758-c286-45e3-aaf6-f0868eed0fea.png)

**How to test**
@christopherferreira9 has build the tools to develop "Session Persistence" on all platforms supported by MM SDK (JS desktop/mobile, iOS, Unity desktop/mobile) with versions of the SDK supporting session persistence and older versions not yet supporting it (to check retro-compatibility)
